spacy_methods.py

```diff
@@ -0,0 +1,213 @@
+
+# Base URL, and use
+url = "https://apnews.com/article/george-santos-federal-charges-updates-33667a0900271e5002459ab748d8fdc8?utm_source=homepage&utm_medium=TopNews&utm_campaign=position_01"
+article_dict = AP_article_dict_builder(url)
+article_txt = AP_article_full_txt(url)
+
+# Initialize the Doc object
+doc = nlp(article_txt)
+
+# Generate list of sentences from Doc object
+def sentence_generator(txt):
+    doc = nlp(txt)
+    sentences = [sent.text for sent in doc.sents]
+    return list(sentences)
+
+
+sentences = sentence_generator(doc)
+
+# Retrieve and store entities
+
+
+def get_specific_entities(sentences):
+    specific_entities = []
+    sentence_count = -1
+    previous_sentence = None
+
+    word_count = 0
+
+    for i, sent in enumerate(sentences):
+        sentence_count += 1
+        doc = nlp(sent)
+
+        for j, ent in enumerate(doc.ents):
+            if ent.text in sent:
+                if ent.label_ in ["GPE", "ORG", "PERSON"]:
+                    start_index = ent.start  # Start index of the entity
+                    end_index = ent.end - 1  # End index of the entity
+                    sentence = ent.sent.text  # The text of the sentence
+                    sentence_start = (
+                        ent.sent.start
+                    )  # Index of the first word in the sentence
+                    sentence_end = (
+                        ent.sent.end - 1
+                    )  # Index of the last word in the sentence
+                    sentence_index = ent.start - sentence_start
+                    word_index = word_count + start_index
+
+                    specific_entities.append(
+                        [
+                            ent.text,
+                            ent.label_,
+                            start_index,
+                            end_index,
+                            sentence,
+                            sentence_count,
+                            sentence_start,
+                            sentence_end,
+                            word_index,
+                        ]
+                    )
+
+            # Adds the sentence_end indexing to the total word count
+        word_count += len(doc)
+    return specific_entities
+
+
+entities = get_specific_entities(sentences)
+
+# raw_entity_list = list(entities)
+
+# Counts the number of distinct times an entity appears in the article.
+def entity_counter(lst):
+
+    raw_entity_list = list(entities)
+    count_dict = {}
+
+    # Index the list elements to a dictionary and increment their counts
+    for i, item in enumerate(lst):
+
+        entity_sentence_check = "{}".format(item[0])
+        if entity_sentence_check in count_dict:
+            count_dict[entity_sentence_check] += 1
+        else:
+            count_dict[entity_sentence_check] = 1
+    return count_dict
+
+
+duplicate_items = entity_counter(entities)
```

Comment:
Same comment about moving it under `if __name__ == "__main__":`

Reply:
Moved. Resolved.